pkg/controller/backup_sync_controller_test.go

```diff
@@ -154,3 +158,133 @@ func TestBackupSyncControllerRun(t *testing.T) {
+			},
+			k8sBackups: []*arktest.TestBackup{
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupA").WithPhase(v1.BackupPhaseCompleted),
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupB").WithPhase(v1.BackupPhaseCompleted),
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupC").WithPhase(v1.BackupPhaseCompleted),
+			},
+			numMacthingBackups: 0,
+		},
+		// {
+		// 	name:      "some overlapping backups",
+		// 	namespace: "ns-1",
+		// 	cloudBackups: []*v1.Backup{
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").Backup,
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").Backup,
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").Backup,
+		// 	},
+		// 	k8sBackups: []*arktest.TestBackup{
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupC").WithPhase(v1.BackupPhaseCompleted),
+		// 	},
+		// numMacthingBackups: 2,
+		// },
+		// {
+		// 	name:      "all overlapping backups",
+		// 	namespace: "ns-1",
+		// 	cloudBackups: []*v1.Backup{
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").Backup,
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").Backup,
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").Backup,
+		// 	},
+		// 	k8sBackups: []*arktest.TestBackup{
+		// 		arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").WithPhase(v1.BackupPhaseCompleted),
+		// 		// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").WithPhase(v1.BackupPhaseCompleted),
+		// 	},
+		// 	numMacthingBackups: 3,
+		// },
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				bs              = &arktest.BackupService{}
+				client          = fake.NewSimpleClientset()
+				sharedInformers = informers.NewSharedInformerFactory(client, 0)
+				logger          = arktest.NewLogger()
+			)
+
+			c := NewBackupSyncController(
+				client.ArkV1(),
+				bs,
+				"bucket",
+				time.Duration(0),
+				test.namespace,
+				sharedInformers.Ark().V1().Backups(),
+				logger,
+			).(*backupSyncController)
+
+			expectedActions := make([]core.Action, 0)
+
+			// setup: insert backups into Kubernetes
+			for _, backup := range test.k8sBackups {
+				actionCreate := core.NewCreateAction(
+					v1.SchemeGroupVersion.WithResource("backups"),
+					test.namespace,
+					backup.Backup,
+				)
+				expectedActions = append(expectedActions, actionCreate)
+
+				actionDelete := core.NewDeleteAction(
```

Comment:
you're not going to expect a delete for **every** k8s backup in each test case -- this is only expected for k8s backups that don't have a corresponding cloud backup.  Need to conditionally setup this expectation

Reply:
Ha. This is the worse offender! Thanks for catching this.